cmd/cluster/sre_operators/listCmd.go

```diff
@@ -0,0 +1,477 @@
+	outdated  bool
+	noHeaders bool
+	operator  string
+
+	genericclioptions.IOStreams
+	kubeCli client.Client
+}
+
+type sreOperator struct {
+	Name     string
+	Current  string
+	Expected string
+	Status   string
+	Channel  string
+}
+
+const (
+	sreOperatorsListExample = `
+	# List SRE operators
+	$ osdctl cluster sre-operators list
+	
+	# List SRE operators without fetching the latest version for faster output
+	$ osdctl cluster sre-operators list --short
+	
+	# List only SRE operators that are running outdated versions
+	$ osdctl cluster sre-operators list --outdated
+	
+	# List a specific SRE operator
+	$ osdctl cluster sre-operators list --operator='OPERATOR_NAME'
+	`
+	sreOperatorsListDescription = `
+	Lists the current version, channel, and status of SRE operators running in the current 
+	cluster context, and by default fetches the latest version from the operators' repositories.
+	
+	A GitLab access token is required to fetch the latest version of the operators, and can be 
+	set in the '.config/osdctl' file as 'gitlab_access: <TOKEN>', or using the 'osdctl setup' command.
+	
+	The command creates a Kubernetes client to access the current cluster context, and GitLab/GitHub
+	clients to fetch the latest versions of each operator from its respective repository.
+	`
+
+	repositoryBranch = "production"
+)
+
+var listOfOperators = []string{
+	"openshift-addon-operator",
+	"aws-vpce-operator",
+	"openshift-custom-domains-operator",
+	"openshift-managed-node-metadata-operator",
+	"openshift-managed-upgrade-operator",
+	"openshift-must-gather-operator",
+	"openshift-ocm-agent-operator",
+	"openshift-osd-metrics",
+	"openshift-rbac-permissions",
+	"openshift-splunk-forwarder-operator",
+	"aws-account-operator",
+	"certman-operator",
+	"openshift-cloud-ingress-operator",
+	"openshift-monitoring",
+	"deadmanssnitch-operator",
+	"openshift-deployment-validation-operator",
+	"gcp-project-operator",
+	"openshift-velero",
+	"pagerduty-operator",
+	"openshift-route-monitor-operator",
+	"openshift-observability-operator",
+}
+
+var listOfOperatorNames = []string{
+	"addon-operator",
+	"aws-vpce-operator",
+	"custom-domains-operator",
+	"managed-node-metadata-operator",
+	"managed-upgrade-operator",
+	"must-gather-operator",
+	"ocm-agent-operator",
+	"osd-metrics-exporter",
+	"rbac-permissions-operator",
+	"splunk-forwarder-operator",
+	"aws-account-operator",
+	"certman-operator",
+	"cloud-ingress-operator",
+	"configure-alertmanager-operator",
+	"deadmanssnitch-operator",
+	"deployment-validation-operator",
+	"gcp-project-operator",
+	"managed-velero-operator",
+	"pagerduty-operator",
+	"route-monitor-operator",
+	"observability-operator",
+}
+
+var (
+	RestoreColor = "\033[0m"
+	Red          = "\033[31m"
+	Bold         = "\u001b[1m"
+	Gap          = "         "
+)
+
+var csv = &unstructured.Unstructured{}
+var csvList = &unstructured.UnstructuredList{}
+var sub = &unstructured.Unstructured{}
+var subList = &unstructured.UnstructuredList{}
```

Comment:
I know you are using `DeepCopy` but we really shouldn't be using global variables, especially when goroutines are involved. Pass these around as dependencies or define them inline of your functions.

Reply:
Thank you for the suggestion, reverted